Lib/test/test_xml_etree.py

```diff
@@ -2588,6 +2591,24 @@ def test_find_xpath(self):
         self.assertRaisesRegex(SyntaxError, 'XPath', e.find, './tag[last()-0]')
         self.assertRaisesRegex(SyntaxError, 'XPath', e.find, './tag[last()+1]')
 
+        # Test concat functionality
+        self.assertEqual(e.find('./tag[@class=concat("d")]').attrib['class'], 'd')
+        self.assertEqual(e.find('./tag[@class=concat("", \'d\', "")]').attrib['class'], 'd')
+        self.assertEqual(e.find('./tag[@value="concat"]').attrib['class'], 'b')
+        self.assertEqual(e.find('./tag[@value=concat("concat")]').attrib['class'], 'b')
+        self.assertEqual(e.find('./tag[@concat="value"]').attrib['class'], 'd')
+        self.assertEqual(e.find('./tag[@concat=concat("value")]').attrib['class'], 'd')
+        self.assertEqual(e.find('./concat[@class=concat("x")]').attrib['class'], 'x')
```

Comment:
Ok, but … none of these tests represents a real use case for `concat()`, does it? What about `concat("prefix-", @attr)` or `concat("'", ., "'")` ? Those seem to be the really interesting cases to me.

I understand that your intention was to allow for escaping, but a) I can't see that being covered by the tests and b) does that really make a use case for `concat()` already?

My fear is that we're opening up a feature here which users will bump into and be disappointed because it doesn't actually do what they want. It really just serves the tiny use case of concatenating string constants right now. That's … unexpected at best.